proxy_manager.go

```diff
@@ -231,3 +232,31 @@ func (m *proxyManager) getFlakeIDGeneratorConfig(name string) FlakeIDGeneratorCo
 func makeProxyName(serviceName string, objectName string) string {
 	return fmt.Sprintf("%s%s", serviceName, objectName)
 }
+
+type proxyDestroyer interface {
+	Destroy(ctx context.Context) error
+}
+
+func (m *proxyManager) destroyProxies(ctx context.Context) {
+	for key, p := range m.proxies {
+		if p == nil {
+			m.serviceBundle.Logger.Log(logger.WeightError, func() string {
+				return fmt.Sprintf("proxy %s key cannot be destroyed, given proxy argument is nil", key)
+			})
+			continue
+		}
+		ds, ok := p.(proxyDestroyer)
+		if !ok {
+			m.serviceBundle.Logger.Log(logger.WeightError, func() string {
+				return fmt.Sprintf("proxy %s key cannot be destroyed, proxy argument does not implement proxyDestroyer", key)
+			})
+			continue
+		}
+		err := ds.Destroy(ctx)
+		if err != nil {
+			m.serviceBundle.Logger.Log(logger.WeightError, func() string {
```

Comment:
How about:
```
m.serviceBundle.Logger.Errorf("proxy %s key cannot be destroyed: %w", key, err)
```

Reply:
Thank, it is better for the verbosity of the logger. [df88887](https://github.com/hazelcast/hazelcast-go-client/pull/772/commits/df8888728cd4a5f4a040179d4df2b0cb77c1c8b6).